### ISSUE TITLE
Unable to make any requests in flutter web

### ISSUE THREAD

**Issue by syky27** [flutterchina/dio#1001]
### New Issue Checklist

- [X] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | web                        |      |
| Platform Version               |   Channel beta, 1.24.0-10.2.pre, on macOS 11.0.1 20B29 darwin-x64, locale en-GB                             |      |
| Dio Version                    |   dio: ^3.0.10                                 |      |
| Android Studio / Xcode Version | Android toolchain - develop for Android devices (Android SDK version 29.0.3) Xcode - develop for iOS and macOS (Xcode 12.2)             |      |
| Repro rate                     | 100%   |      |
| Repro with our demo prj        | No idea          |      |
| Demo project link              | https://github.com/syky27/dio_web |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

I am unable to make any requests while running `flutter web` I saw some other issues performance-wise, but I am unable to make any request.

I created sample project, which works on iOS in Simulator, but does not work in chrome. (see the project link above)

```
The getter 'statusCode' was called on null.
```

Thanks for any help

**Comment by syky27** [comment]
Hmmmm..... I created this sample project because it did not work in my othernon public project, but now I am looking into the console and I see this:
```
Access to XMLHttpRequest at '..../api/Account/Login' from origin 'http://localhost:50371' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

I suppose this could be the issue....

**Comment by syky27** [comment]
CORS, related, sorry.
